doc/how-to-run-in-windows.rst

```diff
@@ -0,0 +1,352 @@
+Haskell, GHC and Cabal in Windows
+=================================
+
+The Windows toolchain and environment
+-------------------------------------
+
+`MinGW <https://www.mingw-w64.org/>`_ is a set of software packages for the
+Windows platform that enable development and compilation targeting the native
+Windows API. MinGW does not offer a complete developer environment so it usually
+is coupled with `MSYS2 <https://www.msys2.org/>`_ which provides many UNIX-like
+tools such as ``bash``, ``make`` or the package manager ``pacman``.
+
+MSYS2 comes with different environments. You can learn more about the different
+environments in the `MSYS2 documentation
+<https://www.msys2.org/docs/environments/>`_. We will be using ``UCRT64`` in the
+examples below but it can be swapped by ``CLANG64`` (as long as the external
+packages you want to use are available for the ``CLANG64`` environment, see
+:ref:`Using system libraries with GHC`).
+
+.. note::
+
+    When the code snippets must be run in a MSYS2 shell, we will use the ``msys2>``
+    prompt and when in a native Windows shell we will use the ``PS>`` or ``cmd>``
+    prompt. If either shell will do, we will just use ``>``.
+
+A small note on configuring and updating MSYS2
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+Before starting with MSYS2, one should make sure it is updated and properly configured. For that:
+
+::
+
+   # Update the system, run twice
+   msys2> pacman --noconfirm -Syuu
+   msys2> pacman --noconfirm -Syuu
+
+   # install basic utilities
+   msys2> pacman --noconfirm -S --needed curl autoconf mingw-w64-ucrt-x86_64-pkgconf
+   msys2> pacman --noconfirm -S ca-certificates
+
+   # Optionally if you like zsh
+   msys2> pacman --noconfirm -S zsh
+
+   # make $HOME the same in both PowerShell and MSYS2's Bash
+   msys2> sed -i -e 's/db_home:.*$/db_home: windows/' /etc/nsswitch.conf
+
+   # Inherit Windows' path in MSYS2
+   msys2> sed -i -e 's/rem set MSYS2_PATH_TYPE=inherit/set MSYS2_PATH_TYPE=inherit/' /c/msys64/msys2_shell.cmd
+
+As a terminal, `Windows Terminal
+<https://apps.microsoft.com/detail/9N0DX20HK701>`_ usually comes in handy. I use
+the following profile:
+
+::
+
+   {
+     "colorScheme": "One Half Dark",
+     "commandline": "C:\\msys64\\msys2_shell.cmd -defterm -here -no-start -ucrt64 -shell zsh",
+     "hidden": false,
+     "icon": "C:\\msys64\\ucrt64.ico",
+     "name": "MSYS2 / UCRT64",
+     "startingDirectory": "C:/Users/<YourName>"
+   }

```

Comment:
This is entirely unnecessary. Cabal documentation does not need to advise the user on which terminal configuration to use, let alone their choice of colour scheme!

Reply:
This was more of a recommendation, thats why it says "I use this" and not "this is the profile that you should use". I was hesitant to add this part but I thought it might be kind for a new user that doesn't really get how the msys2_shell has to be invoked to have an already crafted example.

Anyways, this part is also gone now, I do not mention any terminals. The user can use whatever they want (they also could do it in this version of the document, these were just recommendations).